pkg/queue/bounded_queue_test.go

```diff
@@ -112,6 +93,58 @@ func TestBoundedQueue(t *testing.T) {
 	assert.False(t, q.Produce("x"), "cannot push to closed queue")
 }
 
+// In this test we run a queue with capacity 1 and a single consumer.
+// We want to test the overflow behavior, so we block the consumer
+// by holding a startLock before submitting items to the queue.
+func TestBoundedQueue(t *testing.T) {
+	mFact := metricstest.NewFactory(0)
+	counter := mFact.Counter(metrics.Options{Name: "dropped", Tags: nil})
+
+	q := NewBoundedQueue(1, func(item interface{}) {
+		counter.Inc(1)
+	})
+	assert.Equal(t, 1, q.Capacity())
+
+	var startLock sync.Mutex
+
+	startLock.Lock() // block consumers
+	consumerState := newConsumerState(t)
+
+	q.StartConsumers(1, func(item interface{}) {
```

Comment:
There's still a bunch of boiler plate across these tests. What if the helper function looked like this instead:
```go
func helper(t *testing.T, startConsumers func(q *BounderQueue, consumerFn func(item interface{})))
```

Then the tests become
```go
func TestBoundedQueue(t *testing.T) {
    helper(t, func (q *BounderQueue, consumerFn func(item interface{})) {
        q.StartConsumers(1, consumerFn)
    })
}

func TestBoundedQueueWithFactory(t *testing.T) {
    helper(t, func (q *BounderQueue, consumerFn func(item interface{})) {
        q.StartConsumersWithFactory(1, func () Consumer { return ConsumerFunc(consumerFn) })
    })
}
```